pkg/syncthing/syncthing.go

```diff
@@ -701,33 +670,51 @@ func (s *Syncthing) Restart(ctx context.Context) error {
 	return err
 }
 
-// Stop halts the background process and cleans up.
-func (s *Syncthing) Stop(force bool) error {
-	pidPath := filepath.Join(s.Home, syncthingPidFile)
-	pid, err := getPID(pidPath)
-	if os.IsNotExist(err) {
-		return nil
+// HardTerminate halts the background process and cleans up.
+func (s *Syncthing) HardTerminate() error {
+	pList, err := process.Processes()
+	if err != nil {
+		return err
 	}
-
-	if !force {
-		if pid != s.pid {
-			log.Infof("syncthing pid-%d wasn't created by this command, skipping", pid)
-			return nil
+	for _, p := range pList {
+		name, err := p.Name()
+		if err != nil {
+			log.Infof("error getting name for process %d: %s", p.Pid, err.Error())
+			continue
+		}
+		if !strings.Contains(name, "syncthing") {
+			continue
+		}
+		cmdline, err := p.Cmdline()
+		if err != nil {
+			return err
+		}
+		if !strings.Contains(cmdline, fmt.Sprintf("-home %s", s.Home)) {
+			continue
+		}
+		log.Infof("terminating syncthing %d with wait: %s", p.Pid, s.Home)
+		if err := terminate(p, true); err != nil {
+			log.Infof("error terminating syncthing %d with wait: %s", p.Pid, err.Error())
 		}
+		log.Infof("terminated syncthing %d with wait: %s", p.Pid, s.Home)
 	}
+	return nil
+}
 
-	if err := s.cleanupDaemon(pid, force); err != nil {
-		return err
+// SoftTerminate best effor to halt the background process
+func (s *Syncthing) SoftTerminate() error {
```

Comment:
Both soft and hard Terminate kill the process using the terminate function. What makes one soft vs hard?